source/v4/saml.yaml

```diff
@@ -0,0 +1,182 @@
+  /saml/metadata:
+    get:
+      tags:
+        - SAML
+      summary: Get metadata
+      description: |
+        Get SAML metadata from the server. SAML must be configured properly.
+        ##### Permissions
+        No permission required.
+      responses:
+        '200':
+          description: SAML metadata retrieval successful
+          schema:
+            type: string
+        '501':
+          $ref: '#/responses/NotImplemented'
+
+  /saml/certificate/idp:
+    post:
+      tags:
+        - SAML
+      summary: Upload IDP certificate
+      description: |
+        Upload the IDP certificate to be used with your SAML configuration. This will also set the filename for the IdpCertificateFile setting in your `config.json`.
+        ##### Permissions
+        Must have `manage_system` permission.
+      parameters:
+        - name: certificate
+          in: formData
+          description: The IDP certificate file
+          required: true
+          type: file
+      responses:
+        '200':
+          description: SAML certificate upload successful
+          schema:
+            $ref: '#/definitions/StatusOK'
+        '400':
+          $ref: '#/responses/BadRequest'
+        '401':
+          $ref: '#/responses/Unauthorized'
+        '403':
+          $ref: '#/responses/Forbidden'
+        '501':
+          $ref: '#/responses/NotImplemented'
+
+    delete:
+      tags:
+        - SAML
+      summary: Remove IDP certificate
+      description: |
+        Delete the current IDP certificate being used with your SAML configuration. This will also disable SAML on your system as this certificate is required for SAML.
```

Comment:
I think you're thinking of it from the UI perspective, which we shouldn't do with the API. If this file is removed I can't save the config file unless SAML disabled because it fails the IsValid check. The UI can just re-enable SAML in the config when they upload a new certificate